### PR TITLE
[eas] Created upload command for EAS builds on iOS

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Create a basic command for uploading builds that were made with EAS. I'm not fully convinced that this is the best approach.
- Pending https://github.com/expo/expo-cli/pull/2504

### CODE REVIEW

File: packages/expo-cli/src/commands/eas-build/status/action.ts [hunk 1]
```diff
@@ -7,16 +7,14 @@ import { ensureProjectExistsAsync } from '../../../projects';
 import { printTableJsonArray } from '../../utils/cli-table';
```
Comment (dsokal) [inline packages/expo-cli/src/commands/eas-build/status/action.ts:7]: What's the purpose of the changes in this file? Just refactor?

File: packages/expo-cli/src/commands/upload.ts [hunk 2]
```diff
@@ -44,85 +47,125 @@ export default function (program: Command) {
...
+
+    const args = pick(options, SOURCE_OPTIONS);
+    if (Object.keys(args).length > 1) {
+      throw new Error(`You have to choose only one of: --path, --id, --latest, --url`);
+    }
+    IOSUploader.validateOptions(options);
+    const uploader = new IOSUploader(projectDir, { ...options, useEAS });
+    await uploader.upload();
+  } catch (err) {
+    log.error('Failed to upload the standalone app to the app store.');
```
Comment (dsokal) [inline packages/expo-cli/src/commands/upload.ts:149]:
```suggestion
    log.error('Failed to upload the standalone app to the App Store.');
```

**Review by dsokal — COMMENTED** [review]
I'm not totally against this new command but I think it'd be better to:
- [server side] implement the iOS part of Submission Service,
- [cli] refactor the iOSBuilder so the upload experience is similar to Android uploads (you can try `expo upload:android --use-submission-service` to see how much different it is to iOS uploads), the new code is in `src/commands/upload/submission-service`,
- tell users (and/or add explicit command descriptions) that `upload:ios` and `upload:android` are only for Turtle v1 builds but we'll implement similar commands for EAS builds in the near feature,
- generally don't refactor the legacy code which we planned to get rid of anyway.

The main reason why I don't like this is that in my opinion, we should approach the upload process of new builds more comprehensively and don't tweak the old, unmaintained code. It's better to start with limited functionality - which is EAS Builds without the upload command than to start with all features that we had for old builds but already knowing the experience is not great (and there are probably some bugs).

File: packages/expo-cli/src/commands/eas-build/status/action.ts (hunk 1 above)
Comment (EvanBacon) [reply packages/expo-cli/src/commands/eas-build/status/action.ts:7]: I split the API from the table logging, it's used to query the builds from v2.

**Comment by EvanBacon** [comment]
I'm not sure this the right approach either. The two problems I wanted to explore fixing were the tar.gz extraction and the error handling. Most of this PR is error handling + trouble shooting.

**Comment by dsokal** [comment]
I don't agree that this is just error handling + troubleshooting. You've introduced a new command - `eas:upload` in this PR (and added a feature to fetch v2 builds in IOSBuilder). I'd much better prefer to print a warning that `upload:ios`/`upload:android` are not for EAS Builds so you use it at your own risk.

**Comment by dsokal** [comment]
Also, if I was trying to make the upload commands compatible with EAS Builds I'd start with the Android one - it makes more sense since it's following the design decisions we made with Brent (about how it should look like + online mode with Submission Service + refactor and some tests).
